src/schemaDesigner/schemaDesignerWebviewController.ts

```diff
@@ -188,7 +189,20 @@ export class SchemaDesignerWebviewController extends ReactWebviewPanelController
                         sessionId: this._sessionId,
                     });
                     // Open the document in the editor with the connection
-                    void this.mainController.onNewQuery(this.treeNode, result?.script);
+                    if (this.treeNode) {
```

Comment:
Is this change related to the agent-mode tools?

Reply:
Yes, we want to support `show_schema` tool in agent mode and that uses schema designer.
Currently schema designer is coupled with OE (using treeNode), treeNode is used for two purposes:
1. connectionStr gen for the backend service
2. opening the publish script in a connected query editor (by calling `NewQuery` on a treeNode)

This PR adds an extra code path that takes in a connectionUri and doesn't rely on OE. This change here handles opening the publish script in a new editor and connect to the server/db using connectionUri.